src/blob/core.rs

```diff
@@ -251,8 +250,8 @@ where
         debug!("blob write");
         // Only one upgradable_read lock is allowed at a time
         let (partially_serialized, header) = record.to_partially_serialized_and_header()?;
-        let blob = blob.upgradable_read().await;
-        Self::write_locked(blob, key, partially_serialized, header).await
+        let blob = blob.read().await;
```

Comment:
This change is questionable, because it allows multiple simultaneous writes into a single file. We should estimate pros and cons of this change

Reply:
Can you run benchmark with simultaneous writes from multiple threads into a single blob to compare `read()` vs `upgradable_read()` locks?
I expect that when the write goes to the disk cache, the `read()` lock will be faster, but when the disk cache is full, then `upgradable_read()` will become faster. 

`upgradable_read()` implies that only one thread is writing at a time, and this is generally safer in case of emergencies

Reply:
I've run some tests, using `upgradable_read` compared to master slightly improves read performance, and write performance stays the same. Using `read` compared to `upgradable_read` improves write performance without any noticable degradation in other aspects.

Reply:
Please, replace `RwLock` from `async-lock` crate to `RwLock` from `tokio`, because there is no need to use read lock upgrade feature anymore. To ensure single writer execution at a time, you can add `Semaphore` inside `Blob` structure.

Additionally, please run the test with and without `Semaphore`, running an additional `bobp` instance that writes large (~100Mb) records

Reply:
According to my tests and investigations, it is better to return back to the `RwLock` from `async-lock` crate (revert last 2 commits) and use `upgradable_read`.
Additionally, we should add a comment: `Only one upgradable_read lock is allowed at a time. This is critical because we want to be sure that only one write operation is running at a time.`